components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/SubscriptionsApiServiceImpl.java

```diff
@@ -123,6 +124,27 @@ public Response subscriptionsSubscriptionIdGet(String subscriptionId, String acc
         }
     }
 
+    @Override public Response subscriptionsSubscriptionIdPut(String subscriptionId, SubscriptionDTO body, String accept,
+            String ifNoneMatch, String ifModifiedSince) {
+
+        String username = RestApiUtil.getLoggedInUsername();
+        APIProvider apiProvider = null;
+        APIConsumer apiConsumer = null;
+        try {
+            apiProvider = RestApiUtil.getProvider(username);
+            SubscribedAPI subscribedAPI = SubscriptionMappingUtil.fromDTOToSubscription(body);
+            apiProvider.updateSubscription(subscribedAPI);
+
+            //retrieve the updated Subscription
+            apiConsumer = RestApiUtil.getConsumer(username);
+            SubscribedAPI updatedSubscribedAPI = apiConsumer.getSubscriptionByUUID(subscriptionId);
+            SubscriptionDTO subscriptionDTO = SubscriptionMappingUtil.fromSubscriptionToDTO(updatedSubscribedAPI);
+            return Response.ok().entity(subscriptionDTO).build();
+        } catch (APIManagementException e) {
+            throw new InternalServerErrorException(e);
```

Comment:
Error handling fixed in commit https://github.com/wso2/carbon-apimgt/commit/8049b7e2779abb3dd6bf13aeb70b062438e7c5b7